lib/resourceapply/apps.go

```diff
@@ -29,13 +29,19 @@ func ApplyDeploymentv1(ctx context.Context, client appsclientv1.DeploymentsGette
 		return nil, false, nil
 	}
 
+	var original appsv1.Deployment
```

Comment:
If we keep your change below move these two lines into `if` where it's used.

Reply:
Can't do that, we need to make the original deepcopy of `existing` before it is modified on L35 by `resourcemerge.EnsureDeployment`